packages/compass/src/main/telemetry.ts

```diff
@@ -79,7 +81,8 @@ class CompassTelemetry {
     });
   }
 
-  private static async identify() {
+  // Keep this method synchronous to avoid race conditions.
+  private static identify() {
```

Comment:
@mcasimir I looked into why this was failing, turns out the `fs.readFile` call that `getOsInfo` makes on linux delays the `this.analytics.identify()` call so much that it only happens after the first `.track()` event.